src/software/ai/navigator/navigator.cpp

```diff
@@ -44,56 +49,59 @@ void Navigator::visit(const KickIntent &kick_intent)
 
 void Navigator::visit(const MoveIntent &move_intent)
 {
+    std::vector<Point> path_points;
+
     Point start =
         this->world.friendlyTeam().getRobotById(move_intent.getRobotId())->position();
     Point dest = move_intent.getDestination();
 
     std::vector<Obstacle> obstacles =
         createCurrentObstacles(move_intent.getAreasToAvoid(), move_intent.getRobotId());
 
-    auto path_planner =
-        std::make_unique<ThetaStarPathPlanner>(this->world.field(), obstacles);
-
-    auto path_points = path_planner->findPath(start, dest);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
+            path_planner_->findPath(start, dest, this->world.field(), obstacles));
+    }
+    catch (const std::bad_variant_access &)
+    {
+        // variant either monostate or Curve, so no valid path
+        auto stop = std::make_unique<StopPrimitive>(move_intent.getRobotId(), false);
+        current_primitive = std::move(stop);
+        return;
+    }
 
-    if (path_points)
+    planned_paths.emplace_back(path_points);
+    if ((path_points).size() > 2)
     {
-        planned_paths.emplace_back(*path_points);
-        if ((*path_points).size() > 2)
-        {
-            current_destination = (*path_points)[1];
-            double segment_final_vel =
-                getCloseToEnemyObstacleFactor((*path_points)[1]) *
-                calculateTransitionSpeedBetweenSegments(
-                    (*path_points)[0], (*path_points)[1], (*path_points)[2],
-                    ROBOT_MAX_SPEED_METERS_PER_SECOND *
-                        Util::DynamicParameters::Navigator::transition_speed_factor
-                            .value());
+        current_destination = (path_points)[1];
+        double segment_final_vel =
+            getCloseToEnemyObstacleFactor((path_points)[1]) *
+            calculateTransitionSpeedBetweenSegments(
+                (path_points)[0], (path_points)[1], (path_points)[2],
+                ROBOT_MAX_SPEED_METERS_PER_SECOND *
+                    Util::DynamicParameters::Navigator::transition_speed_factor.value());
```

Comment:
This expression is way too long, could we split it up a bit? Perhaps
```
double final_speed = ROBOT_MAX_SPEED_METERS_PER_SECOND * Util::DynamicParameters::Navigator::transition_speed_factor.value();
double transition_speed = calculateTransitionSpeedBetweenSegments(p0, p1, p2, final_speed);
double segment_final_vel = getCloseToEnemyObstacleFactor((path_points)[1]) * transition_speed;
```
some pseudo-code in there, but you get the idea

Reply:
out of scope for this PR

Reply:
#814

Reply:
Fair enough!